### PR TITLE
Scaffold fixes

### LINKED ISSUE #3563: linq2db.postgresql has borken dependency
Workaround: use linq2db.cli

### LINKED ISSUE #3567: [Scaffold] Incorrect help text for `single-file` option
```
   --single-file                             : generates all code in single file
         type: bool (optional)
         json: code.single-file
         default: false
```

The default value is `true`, at least I had to explicitly change it to `false` in order to get separate files.

_Originally posted by @Powerz in https://github.com/linq2db/linq2db/discussions/3531#discussioncomment-2789567_

### LINKED ISSUE #3568: CLI generates wrong file name if the name of the class was overridden
When I change the name of the class in the interceptor, the CLI generates the correct class, but it puts it in a file with the wrong name (see example below).

```c#
public class SuperInterceptor : LinqToDB.Scaffold.ScaffoldInterceptors
{
    public override void PreprocessEntity(ITypeParser typeParser, EntityModel entityModel)
    {
        if (entityModel.Metadata.Name.ToString() == "dbo.SuperTable")
        {
            entityModel.Class.Name = "AwesomeTable";
            
            // The name of the file containing the generated class is going to be SuperTable instead of AwesomeTable
        }
    }
}
```

Expected file name: `AwesomeTable.cs`.
Actual file name: `SuperTable.cs`.

### Environment details
linq2db version: 4.0.0
Database Server: Postgres
Operating system: macOS
.NET Framework: 6.0.101

### LINKED ISSUE #3569: [Scaffold] Proposal: add the `generated` suffix to the generated model files
Proposal: add the `generated` suffix to the generated model files e.g. `User.generated.cs` instead of `User.cs`. In this case it'd be easier to make the generated files co-exist with their custom partial brothers.

_Originally posted by @Powerz in https://github.com/linq2db/linq2db/discussions/3531#discussioncomment-2789833_

### PR DESCRIPTION
**Pull request by MaceWindu**
- add t4 defaults to CLI help (fix #3567)
- fix errors in sample JSON
- add `--generated-suffix` option to generate files with `.generated.cs` suffix (fix #3569)
- sync file name with main file class name (fix #3568)
- fix `Npgsql` dependencies for T4 nuget (fix #3563)

### CODE REVIEW

File: NuGet/CLI/README.md [hunk 1]
```diff
@@ -314,7 +314,7 @@ This interceptor allows user to specify which .NET type should be used for speci
 ```cs
 // IMPORTANT: this method called only once for each database type
 // ITypeParser inte
-TypeMapping GetTypeMapping(DatabaseType databaseType, ITypeParser typeParser, TypeMapping defaultMapping);
+TypeMapping? GetTypeMapping(DatabaseType databaseType, ITypeParser typeParser, TypeMapping? defaultMapping);
```
Comment (viceroypenguin) [inline NuGet/CLI/README.md:317]: Wasn't this changed by #3566 to be non-nullable? If so, then the readme should be the non-nullable version.

File: Source/LinqToDB.CLI/Template.tt [hunk 2]
```diff
@@ -121,7 +121,7 @@ public class Interceptors : ScaffoldInterceptors
 	/// <param name="typeParser">Type parser to create value for <see cref="TypeMapping.CLRType"/> property from <see cref="Type"/> instance or type name string.</param>
 	/// <param name="defaultMapping">Default type mapping for specified <paramref name="databaseType"/>.</param>
 	/// <returns>Type mapping information for specified <paramref name="databaseType"/>.</returns>
-	public override TypeMapping GetTypeMapping(DatabaseType databaseType, ITypeParser typeParser, TypeMapping defaultMapping)
+	public override TypeMapping? GetTypeMapping(DatabaseType databaseType, ITypeParser typeParser, TypeMapping? defaultMapping)
```
Comment (viceroypenguin) [inline Source/LinqToDB.CLI/Template.tt:124]: Reversing of #3566 ?

File: NuGet/CLI/README.md (hunk 1 above)
Comment (MaceWindu) [reply NuGet/CLI/README.md:317]: I've fixed compilation to not produce warnings, so NRT is back

File: Source/LinqToDB.CLI/Template.tt (hunk 2 above)
Comment (MaceWindu) [reply Source/LinqToDB.CLI/Template.tt:124]: yep, with fix

**Review by viceroypenguin — APPROVED** [review]

**Pull request merged**